cmd/dynamic-networks-controller/networks-controller.go

```diff
@@ -1,5 +1,137 @@
 package main
 
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	v1coreinformerfactory "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+
+	nadclient "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/clientset/versioned"
+	nadinformers "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/informers/externalversions"
+
+	"github.com/maiqueb/multus-dynamic-networks-controller/pkg/config"
+	"github.com/maiqueb/multus-dynamic-networks-controller/pkg/controller"
+	"github.com/maiqueb/multus-dynamic-networks-controller/pkg/logging"
+)
+
+const (
+	ErrorLoadingConfig int = iota
+	ErrorBuildingController
+)
+
 func main() {
-	// TODO implement skeleton
+	klog.InitFlags(nil)
+	configFilePath := flag.String(
+		"config",
+		config.DefaultDynamicNetworksControllerConfigFile,
+		"Specify the path to the multus-daemon configuration")
+
+	flag.Parse()
+
+	controllerConfig, err := config.LoadConfig(*configFilePath)
+	if err != nil {
+		klog.Errorf("failed to load the multus-daemon configuration: %v", err)
+		os.Exit(ErrorLoadingConfig)
+	}
+
+	stopChannel := make(chan struct{})
+
+	podNetworksController, err := newController(stopChannel, controllerConfig)
+	if err != nil {
+		klog.Errorf("failed to instantiate the %s controller: %v", controller.AdvertisedName, err)
+		close(stopChannel) // deferred calls will not be called after os.Exit is called
```

Comment:
Would it make sense to defer it right after line 50, before handling the error?

Reply:
No, `os.Exit` does not respect defer - just shoots the process down.

Reply:
Ah right, brainfart on my end